jgt_common/assert_.py

```diff
@@ -1,132 +1,59 @@
+# NOTE TO IMPLEMENTORS:
+# This module depends on the contracts described in our sibling module ``check``.
+# In the spirit of DRY, please see that module for details.
 
 
-def _msg_concat(prefix, body):
-    """Join with a space if prefix isn't empty."""
-    return "{} {}".format(prefix, body) if prefix else body
+from inspect import isfunction as _isfunction
 
+import wrapt as _wrapt
 
-def not_eq(expected, actual, msg=""):
-    """Assert the values to not be equal."""
-    assert expected != actual, _msg_concat(
-        msg, "Expected '{}' to be not equal to actual '{}'".format(expected, actual)
-    )
+from . import check as _check
 
+_check_module_name = _check.__name__
 
-def eq(expected, actual, msg=""):
-    """Assert the values are equal."""
-    assert expected == actual, _msg_concat(
-        msg, "Expected '{}' == actual '{}'".format(expected, actual)
-    )
+_glbls = globals()
 
 
-def less(a, b, msg=""):
-    """Assert that a < b."""
-    assert a < b, _msg_concat(msg, "Expected '{}' < '{}'".format(a, b))
+# Used internally, but there is no reason to prevent it from being used externally.
+@_wrapt.decorator
+def assert_if_truthy(wrapped, instance, args, kwargs):
+    """Assert if the decorated function returns a truthy value (error indicator)."""
+    result = wrapped(*args, **kwargs)
+    assert not result, result
 
 
-def less_equal(a, b, msg=""):
-    """Assert that a <= b."""
-    assert a <= b, _msg_concat(msg, "Expected '{}' <= '{}'".format(a, b))
-
-
-def greater(a, b, msg=""):
-    """Assert that a > b."""
-    assert a > b, _msg_concat(msg, "Expected '{}' > '{}'".format(a, b))
-
-
-def greater_equal(a, b, msg=""):
-    """Assert that a >= b."""
-    assert a >= b, _msg_concat(msg, "Expected '{}' >= '{}'".format(a, b))
-
-
-def is_in(value, sequence, msg=""):
-    """Assert that value is in the sequence."""
-    assert value in sequence, _msg_concat(
-        msg, "Expected: '{}' to be in '{}'".format(value, sequence)
-    )
-
-
-def any_in(a_sequence, b_sequence, msg=""):
-    """Assert at least one member of a_sequence is in b_sequence."""
-    assert any(a in b_sequence for a in a_sequence), _msg_concat(
-        msg, "None of: '{}' found in '{}'".format(a_sequence, b_sequence)
-    )
+def _pretty_assert_for(fun):
+    """
+    Make fun an asserting function _and_ take ownership of the wrapper.
 
+    wrapt preserves a little _too_ much and we want the wrapped function
+    to have a more accurate doc string and a module attribute that shows it belongs
+    to us instead of ``check``, which could be confusing.
+    """
+    new_fun = assert_if_truthy(fun)
+    new_fun.__doc__ = new_fun.__doc__.replace("Check", "Assert")
+    new_fun.__module__ = __name__
+    return new_fun
 
-def not_in(item, sequence, msg=""):
-    """Assert item is not in sequence."""
-    assert item not in sequence, _msg_concat(
-        msg, "Did NOT Expect: '{}' to be in '{}'".format(item, sequence)
-    )
 
+for _thing_name in dir(_check):
```

Comment:
Nit: What you have here works just fine, but since we already have `inspect` in use, I'm wondering if it might be cleaner to use it all around instead of `dir`:
```
import inspect as _inspect

def _should_be_wrapped(member):
    name, thing = member
    return all([
        not name.startswith("_"),
        _inspect.isfunction(thing),
        _inspect.getmodule(thing).__name__ == _check_module_name
    ])

for (_name, _func) in filter(_should_be_wrapped, _inspect.getmembers(_check)):
    _glbls[_name] = _pretty_assert_for(_func)
```